### DISCUSSION TITLE
Reproducibility with alisim

### DISCUSSION THREAD

**Discussion by tufeiliu** [iqtree/iqtree2#271] [General]
Hello,

I ran a Monte Carlo simulation on 8 protein sequences with 10 sites, using the following command with a fixed seed:

`iqtree2 --alisim output_prefix -T iqtree_nthreads -m best_evolution_model -te gene_tree --root-seq root_node --num-alignments nalign --seed 8`

I tested this command on 10 identical files containing the 8 protein sequences and 10 sites. I observed that the generated .phy files were consistent across these tests. However, when I changed the value of iqtree_nthreads, I still obtained 10 identical .phy files within each set, but the files differed between sets with different thread counts.

Could you explain why this happens, even though the same seed was used for all analyses?

Looking forward to your explanation!
Please let me know if any further adjustments are needed. Many thanks!

**Comment by roblanf** [comment]
Good question! This one is definitely not obvious, and surprises a lot of people. The short answer is that it's not possible for analyses (in any software) to be precisely reproducible with >1 thread. 

A longer answer is provided by @bqminh in the answer to this thread:

https://github.com/iqtree/iqtree2/discussions/233

Rob

**Reply by tufeiliu** [reply]
Thanks Rob! That's a bad news. Apart from that, I have a follow up questions regarding the seeds. I was wondering which random data generator is used by iqtree2? Is it Mersenne Twister?

**Reply by tufeiliu** [reply]
Thanks Rob! That's a bad news. Apart from that, I have a follow up questions regarding the seeds. I was wondering which random data generator is used by iqtree2? Is it Mersenne Twister?
